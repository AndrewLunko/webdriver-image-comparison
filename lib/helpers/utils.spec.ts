import {pathExistsSync, removeSync} from 'fs-extra';
import {join} from 'path';
import {
  checkAndroidChromeDriverScreenshot,
  checkAndroidNativeWebScreenshot,
  checkIsAndroid,
  checkIsIos,
  checkIsMobile,
  checkTestInBrowser, checkTestInMobileBrowser, formatFileName, getAndCreatePath,
} from './utils';
import {FormatFileNameOptions, GetAndCreatePathOptions} from './utils.interfaces';

describe('', () => {

  describe('getAndCreatePath', () => {
    const folder = join(process.cwd(), '/.tmp');

    beforeEach(() => removeSync(folder));

    it('should create the folder and return the folder name for a device that needs to have it\s own folder', () => {
      const options: GetAndCreatePathOptions = {
        browserName: '',
        deviceName: 'deviceName',
        isMobile: true,
        savePerInstance: true,
      };
      const expectedFolderName = join(folder, options.deviceName);

      expect(pathExistsSync(expectedFolderName)).toEqual(false);
      expect(getAndCreatePath(folder, options)).toEqual(expectedFolderName);
      expect(pathExistsSync(expectedFolderName)).toEqual(true);
    });

    it('should create the folder and return the folder name for a browser that needs to have it\s own folder', () => {
      const options: GetAndCreatePathOptions = {
        browserName: 'browser',
        deviceName: '',
        isMobile: false,
        savePerInstance: true,
      };
      const expectedFolderName = join(folder, `desktop_${options.browserName}`);

      expect(pathExistsSync(expectedFolderName)).toEqual(false);
      expect(getAndCreatePath(folder, options)).toEqual(expectedFolderName);
      expect(pathExistsSync(expectedFolderName)).toEqual(true);
    });

    it('should create the folder and return the folder name for a browser', () => {
      const options: GetAndCreatePathOptions = {
        browserName: 'browser',
        deviceName: '',
        isMobile: false,
        savePerInstance: false,
      };

      expect(pathExistsSync(folder)).toEqual(false);
      expect(getAndCreatePath(folder, options)).toEqual(folder);
      expect(pathExistsSync(folder)).toEqual(true);
    });
  });

  describe('formatFileName', () => {
    const formatFileOptions: FormatFileNameOptions = {
      browserName: '',
      deviceName: '',
      devicePixelRatio: 2,
      formatImageName: '',
      isMobile: false,
      isTestInBrowser: true,
      logName: '',
      name: '',
      outerHeight: 768,
      outerWidth: 1366,
      screenHeight: 900,
      screenWidth: 1400,
      tag: 'theTag',
    };

    it('should format a string with all options provided', () => {
      formatFileOptions.formatImageName = '{browserName}-{dpr}-{height}-{logName}-{name}-{tag}-{width}';
      formatFileOptions.browserName = 'chrome';
      formatFileOptions.logName = 'chrome-latest';
      formatFileOptions.name = 'chrome-name';

      expect(formatFileName(formatFileOptions)).toEqual('chrome-2-768-chrome-latest-chrome-name-theTag-1366.png');
    });

    it('should format a string for mobile app', () => {
      formatFileOptions.formatImageName = '{tag}-{mobile}-{dpr}-{width}x{height}';
      formatFileOptions.deviceName = 'iPhoneX';
      formatFileOptions.isMobile = true;
      formatFileOptions.isTestInBrowser = false;

      expect(formatFileName(formatFileOptions)).toEqual('theTag-app-2-1400x900.png');
    });

    it('should format a string for mobile browser', () => {
      formatFileOptions.formatImageName = '{tag}-{mobile}-{dpr}-{width}x{height}';
      formatFileOptions.browserName = 'chrome';
      formatFileOptions.deviceName = 'iPhoneX';
      formatFileOptions.isMobile = true;
      formatFileOptions.isTestInBrowser = true;

      expect(formatFileName(formatFileOptions)).toEqual('theTag-chrome-2-1400x900.png');
    });
  });

  describe('checkIsMobile', () => {
    it('should return false when no platform name is provided', () => {
      expect(checkIsMobile('')).toEqual(false);
    });

    it('should return true when a platform name is provided', () => {
      expect(checkIsMobile('ios')).toEqual(true);
    });
  });

  describe('checkIsAndroid', () => {
    it('should return false when no platform name is provided', () => {
      expect(checkIsAndroid('')).toEqual(false);
    });

    it('should return false when a platform name is provided that is not accepted', () => {
      expect(checkIsAndroid('chrome')).toEqual(false);
    });

    it('should return true when a valid platform name is provided', () => {
      expect(checkIsAndroid('androId')).toEqual(true);
    });
  });

  describe('checkIsIos', () => {
    it('should return false when no platform name is provided', () => {
      expect(checkIsIos('')).toEqual(false);
    });

    it('should return false when a platform name is provided that is not accepted', () => {
      expect(checkIsIos('chrome')).toEqual(false);
    });

    it('should return true when a valid platform name is provided', () => {
      expect(checkIsIos('IoS')).toEqual(true);
    });
  });

  describe('checkTestInBrowser', () => {
    it('should return false when no browser name is provided', () => {
      expect(checkTestInBrowser('')).toEqual(false);
    });

    it('should return true when a browser name is provided', () => {
      expect(checkTestInBrowser('chrome')).toEqual(true);
    });
  });

  describe('checkTestInMobileBrowser', () => {
    it('should return false when no platform name is provided', () => {
      expect(checkTestInMobileBrowser('', 'chrome')).toEqual(false);
    });

    it('should return false when a plaform but no browser name is provided', () => {
      expect(checkTestInMobileBrowser('ios', '')).toEqual(false);
    });

    it('should return true when a plaform and a browser name is provided', () => {
      expect(checkTestInMobileBrowser('ios', 'chrome')).toEqual(true);
    });
  });

  describe('checkAndroidNativeWebScreenshot', () => {
    it('should return false when no platform name is provided', () => {
      expect(checkAndroidNativeWebScreenshot('', false)).toEqual(false);
    });

    it('should return false when iOS and nativeWebscreenshot true is provided', () => {
      expect(checkAndroidNativeWebScreenshot('ios', true)).toEqual(false);
    });

    it('should return false when iOS and nativeWebscreenshot false is provided', () => {
      expect(checkAndroidNativeWebScreenshot('ios', false)).toEqual(false);
    });

    it('should return false when Android and nativeWebscreenshot false is provided', () => {
      expect(checkAndroidNativeWebScreenshot('Android', false)).toEqual(false);
    });

    it('should return true when Android and nativeWebscreenshot true is provided ', () => {
      expect(checkAndroidNativeWebScreenshot('Android', true)).toEqual(true);
    });
  });

  describe('checkAndroidChromeDriverScreenshot', () => {
    it('should return false when no platform name is provided', () => {
      expect(checkAndroidChromeDriverScreenshot('', false)).toEqual(false);
    });

    it('should return false when iOS and nativeWebscreenshot true is provided', () => {
      expect(checkAndroidChromeDriverScreenshot('ios', true)).toEqual(false);
    });

    it('should return false when iOS and nativeWebscreenshot false is provided', () => {
      expect(checkAndroidChromeDriverScreenshot('ios', false)).toEqual(false);
    });

    it('should return false when Android and nativeWebscreenshot true is provided', () => {
      expect(checkAndroidChromeDriverScreenshot('Android', true)).toEqual(false);
    });

    it('should return true when Android and nativeWebscreenshot false is provided ', () => {
      expect(checkAndroidChromeDriverScreenshot('Android', false)).toEqual(true);
    });
  });
});
