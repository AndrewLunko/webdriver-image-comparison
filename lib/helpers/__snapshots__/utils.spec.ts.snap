// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` calculateDprData should multiple all number values by the dpr value 1`] = `
Object {
  "1": 6,
  "a": 2,
  "a1": 18,
  "b": 4,
  "bool": true,
  "string": "string",
}
`;

exports[` checkAndroidChromeDriverScreenshot should return false when Android and nativeWebscreenshot true is provided 1`] = `false`;

exports[` checkAndroidChromeDriverScreenshot should return false when iOS and nativeWebscreenshot false is provided 1`] = `false`;

exports[` checkAndroidChromeDriverScreenshot should return false when iOS and nativeWebscreenshot true is provided 1`] = `false`;

exports[` checkAndroidChromeDriverScreenshot should return false when no platform name is provided 1`] = `false`;

exports[` checkAndroidChromeDriverScreenshot should return true when Android and nativeWebscreenshot false is provided  1`] = `true`;

exports[` checkAndroidNativeWebScreenshot should return false when Android and nativeWebscreenshot false is provided 1`] = `false`;

exports[` checkAndroidNativeWebScreenshot should return false when iOS and nativeWebscreenshot false is provided 1`] = `false`;

exports[` checkAndroidNativeWebScreenshot should return false when iOS and nativeWebscreenshot true is provided 1`] = `false`;

exports[` checkAndroidNativeWebScreenshot should return false when no platform name is provided 1`] = `false`;

exports[` checkAndroidNativeWebScreenshot should return true when Android and nativeWebscreenshot true is provided  1`] = `true`;

exports[` checkIsAndroid should return false when a platform name is provided that is not accepted 1`] = `false`;

exports[` checkIsAndroid should return false when no platform name is provided 1`] = `false`;

exports[` checkIsAndroid should return true when a valid platform name is provided 1`] = `true`;

exports[` checkIsIos should return false when a platform name is provided that is not accepted 1`] = `false`;

exports[` checkIsIos should return false when no platform name is provided 1`] = `false`;

exports[` checkIsIos should return true when a valid platform name is provided 1`] = `true`;

exports[` checkIsMobile should return false when no platform name is provided 1`] = `false`;

exports[` checkIsMobile should return true when a platform name is provided 1`] = `true`;

exports[` checkTestInBrowser should return false when no browser name is provided 1`] = `false`;

exports[` checkTestInBrowser should return true when a browser name is provided 1`] = `true`;

exports[` checkTestInMobileBrowser should return false when a plaform but no browser name is provided 1`] = `false`;

exports[` checkTestInMobileBrowser should return false when no platform name is provided 1`] = `false`;

exports[` checkTestInMobileBrowser should return true when a plaform and a browser name is provided 1`] = `true`;

exports[` formatFileName should format a string for mobile app 1`] = `"theTag-app-2-1400x900.png"`;

exports[` formatFileName should format a string for mobile browser 1`] = `"theTag-chrome-2-1400x900.png"`;

exports[` formatFileName should format a string with all options provided 1`] = `"chrome-2-768-chrome-latest-chrome-name-theTag-1366.png"`;

exports[` getAddressBarShadowPadding should return 0 when this is a check for Android with a native screenshot but without adding a shadow padding 1`] = `0`;

exports[` getAddressBarShadowPadding should return 0 when this is a check for a desktop browser 1`] = `0`;

exports[` getAddressBarShadowPadding should return 0 when this is a check for an Android app 1`] = `0`;

exports[` getAddressBarShadowPadding should return 0 when this is a check for an iOS app 1`] = `0`;

exports[` getAddressBarShadowPadding should return 0 when this is a check for iOS but without adding a shadow padding 1`] = `0`;

exports[` getAddressBarShadowPadding should return 6 when this is a check for Android with a native screenshot and adding a shadow padding 1`] = `6`;

exports[` getAddressBarShadowPadding should return 6 when this is a check for iOS and adding a shadow padding 1`] = `6`;

exports[` getAndCreatePath should create the folder and return the folder name for a browser 1`] = `false`;

exports[` getAndCreatePath should create the folder and return the folder name for a browser 2`] = `true`;

exports[` getAndCreatePath should create the folder and return the folder name for a browser that needs to have its own folder 1`] = `false`;

exports[` getAndCreatePath should create the folder and return the folder name for a browser that needs to have its own folder 2`] = `true`;

exports[` getAndCreatePath should create the folder and return the folder name for a device that needs to have its own folder 1`] = `false`;

exports[` getAndCreatePath should create the folder and return the folder name for a device that needs to have its own folder 2`] = `true`;

exports[` getScreenshotSize should get the screenshot size of a screenshot string with DRP 2 1`] = `
Object {
  "height": 768,
  "width": 1366,
}
`;

exports[` getScreenshotSize should get the screenshot size of a screenshot string with the default DPR 1`] = `
Object {
  "height": 1536,
  "width": 2732,
}
`;

exports[` getToolBarShadowPadding should return 0 when this is a check for Android browser and adding a shadow padding 1`] = `0`;

exports[` getToolBarShadowPadding should return 0 when this is a check for a desktop browser 1`] = `0`;

exports[` getToolBarShadowPadding should return 0 when this is a check for an Android app 1`] = `0`;

exports[` getToolBarShadowPadding should return 0 when this is a check for an Android app with adding a shadow padding 1`] = `0`;

exports[` getToolBarShadowPadding should return 0 when this is a check for an iOS app 1`] = `0`;

exports[` getToolBarShadowPadding should return 0 when this is a check for an iOS app with adding a shadow padding 1`] = `0`;

exports[` getToolBarShadowPadding should return 6 when this is a check for iOS browser and adding a shadow padding 1`] = `6`;
